docs/source/API/core/utilities/device_id.rst

```diff
@@ -0,0 +1,36 @@
+``Kokkos::device_id``
+=====================
+
+.. role:: cppkokkos(code)
+    :language: cppkokkos
+
+Defined in header ``<Kokkos_Core.hpp>``
+
+.. code-block:: cpp
+
+    [[nodiscard]] int device_id() noexcept;
+
+Returns the device id of the device that is used by the Kokkos default device
+backend or ``-1`` if only host backends are enabled.
```

Comment:
Perhaps explicitly say "used by a default-constructed instance of `DefaultDeviceExecutionSpace`"

Reply:
Kokkos only uses one GPU, so all instances are pointing to the same one.

Reply:
I don't think 0-N is useful, because we don't have a user discoverable N, but the sentence is awkward. "used by instances of `DefaultDeviceExecutionSpace`" might be better. We can modify for multiple GPUs later if needed.